client/src/pages/AdminDashboard.js

```diff
@@ -149,65 +191,85 @@ const AdminDashboard = (props) => {
     setAvgHoursToChart(totalVolunteerAvgHoursByEventType);
   }
 
-  function extractVolunteersSignedInByProperty(events, users, uniqueTypes, propName){
-    let result = {};
+  function extractVolunteersSignedInByProperty(
+    events,
+    users,
+    uniqueTypes,
+    propName
+  ) {
+    const result = {};
     let type;
 
-    uniqueTypes.forEach(el => result[el] = parseInt('0'));
-    for (let eventId of users.keys()) {
-      if(propName === 'eventType'){
+    uniqueTypes.forEach((el) => (result[el] = parseInt('0')));
+    for (const eventId of users.keys()) {
+      if (propName === 'eventType') {
         type = events.get(eventId).eventType;
         result[type] = users.get(eventId).length + result[type];
       }
 
-      if(propName === 'hacknight' && typeof events.get(eventId).hacknight !== 'undefined'){
+      if (
```

Comment:
lines less than 100 symbols shouldn't be formatted

Reply:
First, I did a little research and I am not sure on how to set that request specifically. I did want to look at the config settings, but do you have a config setting that you are looking for? https://prettier.io/docs/en/options.html

Secondly, I feel good about these settings, but that may just be me having used formatters for a while now. There is an analogous project in Python now used widely called Black. For some reason, I had assumed that you were already using Prettier in your JS projects, so I did not try and sell it. That said, the Prettier and Black teams do a better job than I can on that. Please take a gander on the video and let me know your thoughts. https://www.youtube.com/watch?v=0Q4kUNx85_4

Reply:
yeah, the video says: *If it fits in one line, stay in one line*
https://youtu.be/0Q4kUNx85_4?t=411

Also, we can try Google eslint config: https://github.com/google/eslint-config-google that uses Google JS style https://google.github.io/styleguide/jsguide.html

Reply:
Yeah, I think that this is part of the Google V. Prettier Auto-formatter issue. It may not exactly relate to ESLint, but it is rated. That same video around 8:40 that states to break on the first argument as part of Rule #2, which has a lot of nuance. There is separate logic and verification on lines 211 and 212, which makes me also agree that we should break them up and put them on separate lines. I also see this example relates pretty well to the part of the video found around 17:15, which states the operating logic is actually on lines 204, and the logic 205-216 is supporting that single truthy statement there. Organizing the code this way shows these lines as a block. Lastly, I can see this relating to the section of the video around 25:00 or so, which talks about bringing prettier onto an existing project. We happen to have formatted the Admin logic, which has a lot of buy in from developers. This I think causes a little more push back than usual, as it feels precious. Even though I state that, I would still highly suggest that we use these settings.

All of this said above, here are the take-aways that I think are super helpful. 
- The code keeps the line on one line for a single method call. It breaks on multiple method calls, or on multiple verification steps. It would not break if there was no && there. 
- The formatting is set to support the validation pathway. This is part of the second on Rule #2 as presented by the video and supports the code block path on 204.
- Formatting can feel frustrating at first. Everyone has their own preferences and automatically formatting a file can feel frustrating as it is a why. I truly think that this goes away with a little time. Also, the rules of this formatter are nuanced as per the video. 
- The higher majority of projects these days are using Prettier. I chose AirBnB and Prettier as it is the most widely adopted settings for React projects such as ours. I think that these settings are pretty good. 
- As far as Google's setting for a project, there is one reason that I did not go the path of setting ESlint to Google. It was due AirBnB's high adoption of React, and Google's overall trends to support Angular. Configuring React with their linter settings required extra work.

Reply:
Most important, we don't need any frustration here at all. This is not a job, everyone is a volunteer. So, no pushing.

The second idea is avoiding overhead. The rule **If it fits in one line, stay in one line** is totally clear for every developer. Going deeper is overengineering. For me first of all. 
Please, don't use multiple lines for function arguments, value assignments, and `if` conditions if it less 100 symbols long.

Reply:
Yeah @alex-anakin, yeah I think the same thing stands that I am not seeing how to meet the goals that you are aiming for with these tools. If you can offer an alternative configuration for the tooling, then I am up for discussing that further. Otherwise, I feel like I have provided more than ample research and evidence on the how and why I chose these tools and configurations. 

For now, I will remove any of the auto-formatting on this file, and any VSCode configurations. This is truly something that I feel passionate about and bring a lot of expertise in, but it's not worth arguing over.

Reply:
Applying the formatting on the file was a great example of what prettier can do. My concern is formatting lines with a length of fewer than 100 symbols. I'm pretty sure there is a rule in prettier settings that controls this action and can be set to false.
Let's merge settings and then try to test eslinter and prettier.